dev/Repeater/SelectionModel.cpp

```diff
@@ -587,16 +587,32 @@ void SelectionModel::SelectImpl(int index, bool select)
 {
     if (m_singleSelect)
     {
-        ClearSelection(true /*resetAnchor*/, false /* raiseSelectionChanged */);
+        // Check if new index is different to current one
+        if (m_rootNode->SelectedIndex() != index)
+        {
+            ClearSelection(true /*resetAnchor*/, false /* raiseSelectionChanged */);
+            auto selected = m_rootNode->Select(index, select);
+            if (selected)
+            {
+                AnchorIndex(winrt::make<IndexPath>(index));
+            }
+            OnSelectionChanged();
+        }
     }
-
-    auto selected = m_rootNode->Select(index, select);
-    if (selected)
+    else
     {
-        AnchorIndex(winrt::make<IndexPath>(index));
+        bool alreadySelected = m_rootNode->IsSelected(index);
+        auto selected = m_rootNode->Select(index, select);
+        if (selected)
+        {
+            AnchorIndex(winrt::make<IndexPath>(index));
```

Comment:
>AnchorIndex [](start = 12, length = 11)

In the case where this index was already selected why are we updating the anchor?

Reply:
Good point.